### ISSUE TITLE
Why when I reasoned, Existence keeps growing

### ISSUE THREAD

**Issue by pipijiev12** [THUDM/ChatGLM2-6B#424]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

First of all, I kept the history and past_key_value through chatglm2 model reasoning, but I only kept 3~5 rounds of history. Logically speaking, this should be that the existing size of cuda will fluctuate steadily after reaching a certain value, instead of accumulating all the time. What is the cause of this. Is this because the network feedback mechanism calculates the loss all the time? After each round of dialogue needs to calculate the loss, the gradient is updated, causing the cuda weight to continue to increase.

### Expected Behavior

When I save a certain number of rounds of history, then model inference will not cause cuda memory to increase

### Steps To Reproduce

First of all, I kept the history and past_key_value through chatglm2 model reasoning, but I only kept 3~5 rounds of history. Logically speaking, this should be that the existing size of cuda will fluctuate steadily after reaching a certain value, instead of accumulating all the time. What is the cause of this. Is this because the network feedback mechanism calculates the loss all the time? After each round of dialogue needs to calculate the loss, the gradient is updated, causing the cuda weight to continue to increase.

### Environment

```markdown
- OS:ubuntu
- Python:3.10.10
- Transformers:4.30.2
- PyTorch:2.0.1
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```
